common/reconciliation/invariant/concrete_execution_exists_test.go

```diff
@@ -42,125 +41,279 @@ import (
+			},
+			wantFixResult: FixResult{
+				FixResultType: FixResultTypeFailed,
+				InvariantName: ConcreteExecutionExists,
+				Info:          "failed to fix: expected current execution",
+			},
+		},
+		{
+			desc: "empty run id - found after current lookup",
+			execution: func() *entity.CurrentExecution {
+				e := getOpenCurrentExecution()
+				e.CurrentRunID = ""
+				return e
+			}(),
+			getConcreteResp: &persistence.IsWorkflowExecutionExistsResponse{Exists: true},
+			getCurrentResp: &persistence.GetCurrentExecutionResponse{
+				RunID: currentRunID,
+			},
+			wantCheckResult: CheckResult{
+				CheckResultType: CheckResultTypeHealthy,
+				InvariantName:   ConcreteExecutionExists,
+			},
+			wantFixResult: FixResult{
+				FixResultType: FixResultTypeSkipped,
+				InvariantName: ConcreteExecutionExists,
+				Info:          "skipped fix because execution was healthy",
+			},
+		},
+		{
+			desc: "empty run id - current lookup returns not found",
+			execution: func() *entity.CurrentExecution {
+				e := getOpenCurrentExecution()
+				e.CurrentRunID = ""
+				return e
+			}(),
+			getCurrentErr: &notExistsError,
+			wantCheckResult: CheckResult{
+				CheckResultType: CheckResultTypeHealthy,
+				InvariantName:   ConcreteExecutionExists,
+				Info:            "current execution does not exist.",
+			},
+			wantFixResult: FixResult{
+				FixResultType: FixResultTypeSkipped,
+				InvariantName: ConcreteExecutionExists,
+			},
+		},
+		{
+			desc: "empty run id - domain name lookup failed",
+			execution: func() *entity.CurrentExecution {
+				e := getOpenCurrentExecution()
+				e.CurrentRunID = ""
+				return e
+			}(),
+			getDomainNameErr: errors.New("error getting domain name"),
+			wantCheckResult: CheckResult{
+				CheckResultType: CheckResultTypeFailed,
+				InvariantName:   ConcreteExecutionExists,
+				Info:            "failed to fetch domainName",
+				InfoDetails:     "error getting domain name",
+			},
+			wantFixResult: FixResult{
+				FixResultType: FixResultTypeSkipped,
+				InvariantName: ConcreteExecutionExists,
+			},
 		},
 	}
-	ctrl := gomock.NewController(s.T())
-	mockDomainCache := cache.NewMockDomainCache(ctrl)
+
 	for _, tc := range testCases {
-		execManager := &mocks.ExecutionManager{}
-		execManager.On("IsWorkflowExecutionExists", mock.Anything, mock.Anything).Return(tc.getConcreteResp, tc.getConcreteErr)
-		execManager.On("GetCurrentExecution", mock.Anything, mock.Anything).Return(tc.getCurrentResp, tc.getCurrentErr)
-		mockDomainCache.EXPECT().GetDomainName(gomock.Any()).Return("test-domain-name", nil).AnyTimes()
-		o := NewConcreteExecutionExists(persistence.NewPersistenceRetryer(execManager, nil, c.CreatePersistenceRetryPolicy()), mockDomainCache)
-		s.Equal(tc.expectedResult, o.Check(context.Background(), tc.execution))
+		t.Run(tc.desc, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			mockDomainCache := cache.NewMockDomainCache(ctrl)
+			execManager := &mocks.ExecutionManager{}
```

Comment:
nit: it might be in the next diff, but could we switch to the mockgen generation of mocks?
Or use --with-expecter on mockery generation for better testability and consistency.